src/applications/personalization/profile360/vet360/components/AddressField/index.jsx

```diff
@@ -16,6 +16,80 @@ import Vet360ProfileField from '../../containers/ProfileField';
 import AddressEditModal from './EditModal';
 import AddressView from './View';
 
+export const inferAddressType = (countryName, stateCode) => {
+  let addressType = ADDRESS_TYPES.DOMESTIC;
+  if (countryName !== USA.COUNTRY_NAME) {
+    addressType = ADDRESS_TYPES.INTERNATIONAL;
+  } else if (ADDRESS_FORM_VALUES.MILITARY_STATES.has(stateCode)) {
+    addressType = ADDRESS_TYPES.OVERSEAS_MILITARY;
+  }
+
+  return addressType;
+};
+
+export const convertNextValueToCleanData = (value) => {
+  const {
+    id,
+    addressLine1,
+    addressLine2,
+    addressLine3,
+    addressPou,
+    city,
+    countryName,
+    stateCode,
+    zipCode,
+    internationalPostalCode,
+    province,
+  } = value;
+
+  const addressType = inferAddressType(countryName, stateCode);
+
+  return {
+    id,
+    addressLine1,
+    addressLine2,
+    addressLine3,
+    addressPou,
+    addressType,
+    city,
+    countryName,
+    province: addressType === ADDRESS_TYPES.INTERNATIONAL ? province : null,
+    stateCode: addressType === ADDRESS_TYPES.INTERNATIONAL ? null : stateCode,
+    zipCode: addressType !== ADDRESS_TYPES.INTERNATIONAL ? zipCode : null,
+    internationalPostalCode: addressType === ADDRESS_TYPES.INTERNATIONAL ? internationalPostalCode : null,
+  };
+};
+
+export const validateCleanData = ({ addressLine1, city, stateCode,  internationalPostalCode, zipCode, countryName }, property) => {
+  const isInternational = inferAddressType(countryName, stateCode) === ADDRESS_TYPES.INTERNATIONAL;
+  const validateAll = !property;
+
+  return {
+    addressLine1: (property === 'addressLine1' || validateAll) && !addressLine1 ? 'Street address is required' : '',
+    city: (property === 'city' || validateAll) && !city ? 'City is required' : '',
+    stateCode: (property === 'stateCode' || validateAll) && !isInternational && !stateCode ? 'State is required' : '',
```

Comment:
Is state displayed as a select? If not, should the state code value be validated, i.e. checked from a list of valid options?

Reply:
thanks! yes these are loaded from a constants file for a list of valid options in a dropdown select

Reply:
Thanks for explaining, it seems like the forms library could be helpful here eventually.